src/server/editorServices.ts

```diff
@@ -717,59 +740,38 @@ namespace ts.server {
         }
 
         private assignScriptInfoToInferredProjectIfNecessary(info: ScriptInfo, addToListOfOpenFiles: boolean): void {

```

Comment:
At two of the callsite to this function, `info.containingProjects.length === 0` was just checked and `addToListOfOpenFiles` is hardcoded to `false`. At the only other callsite, `addToListOfOpenFiles` is hardcoded to `true`. So I think this could be split into 2 different functions that share some common functionality.